src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,75 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Object clazzImplInstance = null;
+        Class<?> clazz = findImpl(interfaceClazz);
```

Comment:
Do we need to check ?
```suggestion
        Class<?> clazz = findImpl(interfaceClazz);
        if (!clazz.isAnnotationPresent(Component.class)) { 
```

Reply:
yeah, the task asks us to do this, or u mean that I should change it and do it differently?

"If the `Component` annotation is missing above the class, we shouldn't be able to create an instance of this class in Injector, and we should throw an exception"

Reply:
feels like this is the simplest way to stop the process of creating new instances for classes that are not permitted by annotation

Reply:
It's better to check `isAnnotationPresent(Component.class)` after `Class<?> clazz = findImpl(interfaceClazz);`
to stop the process and don't implement next logic

Reply:
resolved